File: app/javascript/channels/game_channel.js
import consumer from "./consumer"

consumer.subscriptions.create("GameChannel", {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log("subscribed to channel")
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
    console.log("disconnected from channel")
  },

  received(data) {
    console.log("received data:" + data.game.id)
    // Called when there's incoming data on the websocket for this channel
  }
});

File: app/channels/game_channel.rb
class GameChannel < ApplicationCable::Channel
  def subscribed
    stream_for 'game_channel'
  end

  def speak(data)
    passcode = data['game']['passcode']
    game = Game.where("passcode = ?", passcode).order(:created_at).last
    socket = { game: game }
    GameChannel.broadcast_to('game_channel', socket)
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end
end
